azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -748,21 +630,19 @@ def on_timeout():
 
             # Send the op down, but intercept the return of the op so we can
             # remove the timer when the op is done
+            op.add_callback(self._clear_timer)
             logger.debug("{}({}): Sending down".format(self.name, op.name))
-            self.send_op_down_and_intercept_return(
-                op=op, intercepted_return=self._on_intercepted_return
-            )
+            self.send_op_down(op)
         else:
             self.send_op_down(op)
 
     @pipeline_thread.runs_on_pipeline_thread
-    def _on_intercepted_return(self, op, error):
+    def _clear_timer(self, op, error):
         # When an op comes back, delete the timer and pass it right up.
         if op.timeout_timer:
             op.timeout_timer.cancel()
             op.timeout_timer = None
         logger.debug("{}({}): Op completed".format(self.name, op.name))
```

Comment:
>logger.debug("{}({}): Op completed".format(self.name, op.name)) [](start = 8, length = 63)

this debug string seems like noise.  I would maybe change it to "cancelling timer" and move it into the if block above.

Reply:
Good catch

---
In reply to: [345998377](https://github.com/Azure/azure-iot-sdk-python/pull/392#discussion_r345998377) [](ancestors = 345998377)